### PR TITLE
Fix `zstd` semver comparison

### PR DESCRIPTION
**Pull request by basilenouvellet**
~~We use the `semver.clean` method before comparing the `zstd` version, which strips out the `v` prefix.~~

~~We must compare with a "pure" version number (= without the `v` prefix).~~

Currently, it seems that [all comparison succeed](https://github.com/MansaGroup/mansa/runs/5050929354?check_suite_focus=true#step:21:11), [even with a sufficient version](https://github.com/MansaGroup/mansa/runs/5050929354?check_suite_focus=true#step:4:51).

EDIT: The issue is that `semver.clean(rawVersion)` is not capable of finding the version.
- the raw string is `*** zstd command line interface 64-bits v1.4.4, by Yann Collet ***`
- `semver.clean("*** zstd command line interface 64-bits v1.4.4, by Yann Collet ***")` returns `null`

We need to use our own regexp to parse the version.

### CODE REVIEW

**Review by jeremylvln — DISMISSED** [review]

File: src/tar-utils.ts [hunk 1]
```diff
@@ -20,11 +22,17 @@ async function getTarCompressionMethod(): Promise<CompressionMethod> {
       silent: true,
     })
     .then((out) => out.stdout.trim())
-    .then((out) => [out, semver.clean(out)]);
+    .then((out) => {
+      const extractedVersion = /(v\d+(?:\.\d+){0,})/.exec(out);
```
Comment (basilenouvellet) [inline src/tar-utils.ts]: let's not capture the `v` prefix in the regexp if we compare with `1.3.2` (without the `v` aswell)?

File: src/tar-utils.ts [hunk 2]
```diff
@@ -20,11 +22,17 @@ async function getTarCompressionMethod(): Promise<CompressionMethod> {
...
+      return [out, extractedVersion ? semver.clean(extractedVersion[1]) : null];
+    });
```
Comment (BasileNouvellet) [inline src/tar-utils.ts:28]: no need to clean IMO

```suggestion
    .then((out) => {
      const [_input, extractedVersion] = /v(\d+(?:\.\d+){0,})/.exec(out);
      return [out, extractedVersion];
    });
```

File: src/tar-utils.ts [hunk 3]
```diff
@@ -20,11 +22,17 @@ async function getTarCompressionMethod(): Promise<CompressionMethod> {
       silent: true,
     })
     .then((out) => out.stdout.trim())
-    .then((out) => [out, semver.clean(out)]);
+    .then((out) => {
+      const extractedVersion = /(v\d+(?:\.\d+){0,})/.exec(out);
+      return [out, extractedVersion ? semver.clean(extractedVersion[1]) : null];
```
Comment (basilenouvellet) [inline src/tar-utils.ts]: also, no need to `semver.clean` IMO

File: src/tar-utils.ts (hunk 1 above)
Comment (jeremylvln) [reply src/tar-utils.ts]: Yup totally make sense. Edited the history.

**Review by basilenouvellet — COMMENTED** [review]
Good for me (can't approve because I'm the creator of the PR)

**Pull request merged**